beta/src/pages/learn/adjusting-effect-dependencies.md

```diff
@@ -0,0 +1,1345 @@
+
+```js
+function Page() {
+  const obj = {}; // 🔴 Inline object will be different on every render
+  function fn() { // 🔴 Inline function will be different on every render
+    // ...
+  }
+  useEffect(() => {
+    fn(obj);
+  }, [obj, fn]);
+  // ...
+}
+```
+
+Inline objects and functions are always "new": [`{} !== {}`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality#comparing_objects) and `function(){} !== function(){}`. This makes the `fn` and `obj` dependencies above "always different," so the Effect will re-run after every render. To fix this, you can usually replace object and function dependencies with simpler primitive dependencies, or remove the need for them altogether. You'll learn how to do this [later on this page.](#how-to-fix-an-effect-that-re-runs-too-often)
+
+</DeepDive>
+
+## How to fix an Effect that re-runs too often? {/*how-to-fix-an-effect-that-re-runs-too-often*/}
+
+When your Effect uses a reactive value, you must include it in the dependencies. This may cause problems:
+
+* Sometimes, you want to re-execute *different parts* of your Effect under different conditions.
+* Sometimes, you want to only read *the latest value* of some dependency instead of "reacting" to its changes.
+* Sometimes, a dependency may change too often *unintentionally* because it's an object or a function.
+
+After you adjust the Effect's dependencies to reflect the code, you should always look at the dependency list. Does it make sense for the Effect to re-run when these dependencies change? Sometimes, the answer is "no."
+
+**When this happens, your first instinct might be to omit some dependencies from the list, but that leads to subtle bugs that are very hard to diagnose. Instead, edit the *rest* of the code to *not need* that dependency.** It's not always obvious how to do this, so the rest of this page will introduce you to the most common scenarios.
+
+### Removing an Effect {/*removing-an-effect*/}
+
+The first thing you should think about is whether this code should be an Effect at all. For example, suppose you have a form thats submits a POST request and shows a notification [toast](https://uxdesign.cc/toasts-or-snack-bars-design-organic-system-notifications-1236f2883023). You trigger the Effect by setting state:
+
+```js {4-10}
+function Form() {
+  const [submitted, setSubmitted] = useState(false);
+
+  useEffect(() => {
+    if (submitted) {
+      // 🔴 Avoid: Event-specific logic inside an Effect
+      post('/api/register');
+      showToast('Successfully registered!');
+    }
+  }, [submitted]); // ✅ All dependencies declared
+
+  function handleSubmit() {
+    setSubmitted(true);
+  }  
+
+  // ...
+}
+```
+
+Later, you want to style the notification toast according to the current theme, so you read the current theme. Since `theme` is declared in the component body, it is a reactive value, and you must declare it as a depedency:
```

Comment:
```suggestion
Later, you want to style the notification toast according to the current theme, so you read the current theme. Since `theme` is declared in the component body, it is a reactive value, and you must declare it as a dependency:
```